### DISCUSSION TITLE
Inserting elements in PyAT

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#409] [Python AT]
I am working on FCCee_t_512_nosol_8 lattice in PyAT and i want to insert BPMs and correctors after each Quadrpole in the lattice,
Starting with the PBMs i used the following:

quad_indexes = get_refpts(ring,at.elements.Quadrupole)
for i in quad_indexes:
    BPM = elements.Monitor('BPM', K = 0.0)
    ring.insert(i+1, BPM)

But the results of this is not as i expected, for example the new BPMs indices are = [4,6,8,10 ...]
and the new Quadrupoles indexes are= [3,7,11,14,.....] So the BPMs are not after each Quadrupole. I think the problem is that when i insert the first BPM the quadrupole indices changed so i tried to add a line inside the for loop 
: quad_indexes = get_refpts(ring,at.elements.Quadrupole)

, however the results are not as i expected.

@lfarv @swhite2401

**Comment by lfarv** [comment]
@elafmusa: you are right, when inserting the 1st BPM, the indices of the following elements are changed. The easiest way to solve that is to run your loop backward, starting from the last index.

**Comment by lfarv** [comment]
try this, maybe:
```python
for iq in reversed(quad_indexes):
    BPM = elements.Monitor('BPM')
    ring.insert(iq+1, BPM)
```
